scanpy/preprocessing/_scale.py

```diff
@@ -113,6 +113,13 @@ def scale(
         data, zero_center=zero_center, max_value=max_value, copy=copy, mask_obs=mask_obs
     )
 
+@numba.njit(parallel=True)
+def clip_array(X:np.ndarray | DaskArray ,max_value:float | None = None):
+	a_min, a_max = -max_value, max_value
+	for (r,c) in numba.pndindex(X.shape):
+		X[r,c] = a_max if X[r,c]>a_max else X[r,c]
+		X[r,c] = a_min if X[r,c]<a_min else X[r,c]
```

Comment:
``` suggestion
       if X[r,c] > a_max:
            X[r,c] = a_max
       elif X[r,c] < a_min:
            X[r,c] = a_min
```
This is even faster, because you safe some if statements for me it's 2X faster than your implementation.